File: C/easy/maximum_subarray.c
#define max(a, b) a > b ? a : b

int maxSubArray(int* nums, int n){
    int sum = nums[0], answer = nums[0];
    for(int i=1; i<n; i++){
        sum += nums[i];
        if(sum < nums[i]) sum = nums[i];
        answer = max(answer, sum);
    }
    return answer;
}

File: Python/easy/maximum_subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        if nums == []:
            return 0
        if len(nums) == 1:
            return nums[0]
        
        max_ = -float('inf') 
        sum_ = 0
        for i in nums:
            if i > (sum_+i):
                sum_ = i
            else:
                sum_ += i
            if sum_ > max_:
                max_ = sum_    
        return max_
